apollo-router/src/query_planner/caching_query_planner.rs

```diff
@@ -530,8 +561,11 @@ pub(crate) struct CachingQueryKey {
     pub(crate) hash: Arc<QueryHash>,
     pub(crate) metadata: CacheKeyMetadata,
     pub(crate) plan_options: PlanOptions,
+    pub(crate) config_mode: ConfigMode,
 }
 
+// Update this key every time the cache key or the query plan format has to change
+const CACHE_KEY_VERSION: usize = 0;
 const FEDERATION_VERSION: &str = std::env!("FEDERATION_VERSION");
```

Comment:
Since this value is coming from environment variable maybe it's worth to sanitize it for `:` symbols before putting it in Redis key.
Not a big issue since version shouldn't contain `:`, but maybe worth to do it just in case.

Reply:
this environment variable is set by the build script: https://github.com/apollographql/router/blob/dev/apollo-router/build/main.rs#L34